File: Parrot.cpp
#include <iostream>  // Biblioteka do obsługi wejścia/wyjścia
#include <vector>    // Biblioteka do użycia dynamicznych tablic (wektorów)
#include <cstdlib>   // Biblioteka do generowania liczb losowych
#include <ctime>     // Biblioteka do uzyskiwania czasu systemowego

using namespace std;

class Parrot { // Definicja klasy Parrot
    vector<string> phrases; // Zbiór fraz, które zna papuga

public:
    // Konstruktor z pierwszą frazą
    Parrot(string initialPhrase) {
        phrases.push_back(initialPhrase); // Dodanie początkowej frazy do zbioru
        srand(time(0));  // Ustawienie generatora liczb losowych
    }

    // Metoda dodająca nową frazę do zbioru
    void addPhrase(string newPhrase) {
        phrases.push_back(newPhrase);
    }

    // Metoda wypowiadająca losową frazę z określoną liczbą powtórzeń
    void say(int n) {
        for (int i = 0; i < n; i++) {
            int randIdx = rand() % phrases.size();
            cout << phrases[randIdx] << endl;
        }
    }
};

int main() {
    // Utworzenie obiektu klasy Parrot z początkową frazą
    Parrot parrot("Jestem papugą");
    // Dodawanie nowych fraz do zbioru
    parrot.addPhrase("Test test");
    parrot.addPhrase("Raz dwa trzy");
    parrot.addPhrase("Lubię latać");
    parrot.addPhrase("Witam");

    parrot.say(10);

    return 0;
}

File: Parrot.java
import java.util.ArrayList; // Import biblioteki do dynamicznych tablic
import java.util.Random;    // Import biblioteki do generowania liczb losowych

public class Parrot {
    private ArrayList<String> phrases; // Zbiór fraz, które zna papuga
    private Random random;

    // Konstruktor z pierwszą frazą
    public Parrot(String initialPhrase) {
        phrases = new ArrayList<>();
        phrases.add(initialPhrase); // Dodanie początkowej frazy do zbioru
        random = new Random();
    }

    // Metoda dodająca nową frazę do zbioru
    public void addPhrase(String newPhrase) {
        phrases.add(newPhrase);
    }

    // Metoda wypowiadająca losową frazę z określoną liczbą powtórzeń
    public void say(int n) {
        for (int i = 0; i < n; i++) {
            int randIdx = random.nextInt(phrases.size());
            System.out.println(phrases.get(randIdx));
        }
    }

    public static void main(String[] args) {
        // Utworzenie obiektu klasy Parrot z początkową frazą
        Parrot parrot = new Parrot("Jestem papugą");
        // Dodawanie nowych fraz do zbioru
        parrot.addPhrase("Test test");
        parrot.addPhrase("Raz dwa trzy");
        parrot.addPhrase("Lubię latać");
        parrot.addPhrase("Witam");

        parrot.say(10);
    }
}
